source/common/secret/sds_api.cc

```diff
@@ -144,14 +148,16 @@ void SdsApi::onConfigUpdateFailed(Envoy::Config::ConfigUpdateFailureReason reaso
   init_target_.ready();
 }
 
-void SdsApi::validateUpdateSize(int num_resources) {
+bool SdsApi::validateUpdateSize(int num_resources) {
   if (num_resources == 0) {
-    throw EnvoyException(
-        fmt::format("Missing SDS resources for {} in onConfigUpdate()", sds_config_name_));
+    ENVOY_LOG_MISC(debug, "Missing SDS resource for {} in onConfigUpdate()", sds_config_name_);
+    init_target_.ready();
```

Comment:
Why this call?

Reply:
Again, modified as per the RDS handling.

Reply:
This would mean that receiving an empty SDS update would complete cluster warming, which I'm not sure if we want? I can't speak to why RDS does this, but this seems like an unnecessary change to make if we're just trying to have SDS support resource removals

cc @adisuissa to see if this makes sense from a general xDS implementation pov

Reply:
@snowp Sorry for the rate reply & thanks for the feedback!
So, do you suggest that I should discard this PR ?

I am new to the envoy but would be happy to contribute with proper handling of the resource removal.
@adisuissa, If you already working on it then please share the details & can start with the smaller task.